addon/helpers/route-idle.js

```diff
@@ -1,17 +1,32 @@
 import Helper from '@ember/component/helper';
+import { inject as service } from '@ember/service';
 import { tracked } from '@glimmer/tracking';
 import { whenRouteIdle } from '../scheduler';
 
 export default class RouteIdle extends Helper {
+  @service router;
+
   @tracked isIdle = false;
 
+  constructor() {
+    super(...arguments);
+
+    this.router.on('routeWillChange', () => {
+      if (!this.isDestroying) {
+        this.isIdle = false;
+      }
+    });
```

Comment:
Hmm, I think this may have issues when there are many `route-idle` invocations on a site at once. It also (currently) never tears down so we end up with many subscriptions queued up over time.

Reply:
Refactored to use auto-tracking. Source of truth `isIdle` Boolean is now managed in the same place as the core `route-idle` logic in `scheduler.ts`, so multiple invocations / subscription cleanup should no longer be an issue.